src/is_intersection_empty.jl

```diff
@@ -595,3 +596,83 @@ function is_intersection_empty(set::AbstractPolytope{N},
                               )::Union{Bool, Tuple{Bool, Vector{N}}} where {N<:Real}
     return is_intersection_empty(point, set, witness)
 end
+
+# --- Hyperplane ---
+
+"""
+    is_intersection_empty(X::LazySet{N},
+                          hp::Union{Hyperplane{N}, Line{N}},
+                          [witness]::Bool=false
+                         )::Union{Bool, Tuple{Bool,Vector{N}}} where N<:Real
+
+Check whether a compact set an a hyperplane do not intersect, and otherwise
+optionally compute a witness.
+
+### Input
+
+- `X`       -- compact set
+- `hp`      -- hyperplane
+- `witness` -- (optional, default: `false`) compute a witness if activated
+
+### Output
+
+* If `witness` option is deactivated: `true` iff ``P ∩ Q = ∅``
+* If `witness` option is activated:
+  * `(true, [])` iff ``P ∩ Q = ∅``
+  * `(false, v)` iff ``P ∩ Q ≠ ∅`` and ``v ∈ P ∩ Q``
+
+### Notes
+
+We assume that `X` is compact.
+Otherwise, the support vector queries may fail.
+
+### Algorithm
+
+A compact convex set intersects with a hyperplane iff the support function in
+the negative resp. positive direction of the hyperplane's normal vector ``a`` is
+to the left resp. right of the hyperplane's constraint ``b``:
+
+```math
+-ρ(-a) ≤ b ≤ ρ(a)
+```
+
+For witness generation, we compute a line connecting the support vectors to the
+left and right, and then take the intersection of the line with the hyperplane.
+We follow
+[this algorithm](https://en.wikipedia.org/wiki/Line%E2%80%93plane_intersection#Algebraic_form)
+for the line-hyperplane intersection.
+"""
+function is_intersection_empty(X::LazySet{N},
+                               hp::Union{Hyperplane{N}, Line{N}},
+                               witness::Bool=false
+                              )::Union{Bool, Tuple{Bool,Vector{N}}} where N<:Real
+    sv_left = σ(-hp.a, X)
+    if -dot(sv_left, -hp.a) <= hp.b
+        sv_right = σ(hp.a, X)
+        empty_intersection = (hp.b > dot(sv_right, hp.a))
+    else
+        empty_intersection = true
+    end
+    if witness
+        if empty_intersection
+            v = N[]
+        else
+            point_hp = an_element(hp)
+            point_line = sv_left
+            dir_line = sv_right - sv_left
+            d = dot((point_hp - point_line), hp.a) / dot(dir_line, hp.a)
```

Comment:
maybe use `dir_hyperplane` or `dir_hp` here and above

Reply:
i was suggesting to give a variable name to `hp.a`. for readability of the code. not a big deal.